spec/lib/dfe/bigquery/application_metrics_spec.rb

```diff
@@ -67,14 +70,18 @@
       ]
     end
 
-    before do
+    let(:stub_bigquery_response) do
       stub_bigquery_application_metrics_request(rows: [[
         { name: 'cycle_week', type: 'INTEGER', value: '7' },
         { name: 'nonsubject_filter', type: 'STRING', value: '25 to 29' },
         { name: 'number_of_candidates_submitted_to_date', type: 'INTEGER', value: '100' },
       ]])
     end
 
+    before do
+      stub_bigquery_response
+    end
+
```

Comment:
I've moved the stubbing to memoization because there is a before block at the top of the file that runs before every example. By memoizing them, we can override that before it runs.

Reply:
I do feel like we have to jump around the file a lot to understand the tests.

Reply:
Yes, I too am deeply troubled by this file

Reply:
Sorry, only say this. Nah, I think it just is what it is for now. Let's just write more explicit, readable tests in the future.